src/layout/verticalPanel/allApplicationList.ts

```diff
@@ -0,0 +1,194 @@
+/** Gnome libs imports */
+import * as Clutter from 'clutter';
+import * as GObject from 'gobject';
+import * as Shell from 'shell';
+import { registerGObjectClass } from 'src/utils/gjs';
+import * as St from 'st';
+import { SearchResultEntry } from './searchResultList';
+const ParentalControlsManager = imports.misc.parentalControlsManager;
+
+/** Extension imports */
+const Me = imports.misc.extensionUtils.getCurrentExtension();
+
+@registerGObjectClass
+export class AllApplicationList extends St.BoxLayout {
+    static metaInfo: GObject.MetaInfo = {
+        GTypeName: 'AllApplicationList',
+        Signals: {
+            'result-selected-changed': {
+                param_types: [SearchResultEntry.$gtype],
+                accumulator: 0,
+            },
+        },
+    };
+    disconnectList: Function[] = [];
+    parentalControlsManager;
+    entrySelected: SearchResultEntry | null = null;
+    constructor(searchEntry: St.Entry) {
+        super({
+            style_class: 'search-result-list',
+            vertical: true,
+        });
+        // Note: Clutter typedefs seem to be incorrect. According to the docs `ev` should be a Clutter.KeyEvent, but it actually seems to be a Clutter.Event.
+        const keyId = searchEntry.clutter_text.connect(
+            'key-press-event',
+            this.onKeyPress.bind(this)
+        );
+        this.disconnectList.push(() => searchEntry.disconnect(keyId));
+        this.parentalControlsManager = ParentalControlsManager.getDefault();
+        const parentalId = this.parentalControlsManager.connect(
+            'app-filter-changed',
+            this.updateResults.bind(this)
+        );
+        this.disconnectList.push(() =>
+            this.parentalControlsManager.disconnect(parentalId)
+        );
+
+        this.connect('destroy', () =>
+            this.disconnectList.forEach((disconnectFunction) =>
+                disconnectFunction()
+            )
+        );
+
+        const appSystem = Shell.AppSystem.get_default();
+        appSystem.connect('installed-changed', this.updateResults.bind(this));
+        this.updateResults();
+    }
+
+    get resultEntryList() {
```

Comment:
Personally I prefer avoiding things like `list` in names, and instead indicating this by using plural. E.g. `resultEntries`.

Reply:
I do prefer clarify what kind of container we are talking about because several elements can be grouped into List / Map / Set etc and I find it's more clearer when you have some code that uses both the list and elements of the list
```
// The only difference here is a single letter
for (const element of elements) {}

// I do prefer this
for (cont element of elementList) {}
```